### PR TITLE
[flutter_plugin_tools] Test and comment Dart analysis

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds a unit test and comments intended to avoid accidental breakage of
the Dart repo's run of analysis against this repository.

Addresses https://github.com/flutter/plugins/pull/4183#issuecomment-885767597

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@devoncarew How does this look?

**Comment by devoncarew** [comment]
This looks good - I'm guessing that this will help in terms of reducing upstream breakages (or letting people know when we'll need to update things upstream).

I wonder if a good end state would be to allow a bare run of `dart plugin_tools analyze` to just work on the `flutter/plugins` (and `flutter/packages`) repos? We wouldn't have to know about specific analyze scripts in the repos, or, specific values of flags to pass in.

**Comment by stuartmorgan** [comment]
> I wonder if a good end state would be to allow a bare run of `dart plugin_tools analyze` to just work on the `flutter/plugins` (and `flutter/packages`) repos?

What is `dart plugin_tools`?

File: script/configs/custom_analysis.yaml [hunk 1]
```diff
@@ -8,6 +8,11 @@
 # from a top-level package into more specific packages in order to incrementally
 # migrate a federated plugin.
 #
+# DO NOT move or delete this file without updating
+# https://github.com/dart-lang/sdk/blob/master/tools/bots/flutter/analyze_flutter_plugins.sh
+# which references this file from source, but out-of-repo.
+# Contact stuartmorgan or devoncarew for assistance if necessary.
+
```
Comment (gaaclarke) [inline script/configs/custom_analysis.yaml:15]: Wouldn't it make more sense to have this warning in the cirrus file next to the invocation there?

Comment (stuartmorgan) [reply script/configs/custom_analysis.yaml:15]: Good thought, that's more likely to be noticed in review. I left this longer explanation here, and added a shorter warning to .cirrus.yml that references it.

**Review by gaaclarke — APPROVED** [review]
LGTM. This is definitely an improvement.  I would have rather seen the cirrus script and the dart script share invocations so they can both be updated from this repo.   Hopefully it wont be necessary after all this cleanup though.

**Comment by devoncarew** [comment]
> What is `dart plugin_tools`?

Ah, sorry, I'd meant flutter_plugin_tools, the tool in script/tool (I'd seen 'plugin_tool' from https://github.com/flutter/plugins/blob/master/script/common.sh#L11 but forget it was a bash function).

**Comment by stuartmorgan** [comment]
The long term goal is not too have anything on custom analysis files, as previously discussed. I'm not sure how a totally bare invocation would work though; you added the SDK flag specifically for this use (IIRC because you preferred that to adjusting PATH, which would be the other way to run it).

**Pull request merged**